README.md

```diff
@@ -65,9 +65,12 @@ When you have implemented the tool, answer the questions below, commit it to Git
 ## Questions
 
 How does your method for extracting features work?
+-  We parse each line in the input bed file and add them to a table that is index by the chromosome name. Then for each query we look at the bedlines of the specified chromosome and print them to our outfile if they have a chromosome position between the query's start and end.
 
-What is the complexity of the algorithm, as a function of the size of the two input files? When you answer this, you need to know that you can get the list of chromosomse from a `query.Table` in constant time, but it does, of course, take longer to run through all the lines in it.
+What is the complexity of the algorithm, as a function of the size of the two input files? When you answer this, you need to know that you can get the list of chromosomes from a `query.Table` in constant time, but it does, of course, take longer to run through all the lines in it.
+- Number of bedlines = n, number of querys = m. Parsing every line in the input as well as adding them to a dictionary runs in O(n). Running through all the querys is in O(m) and inside this loop we look at the chromosome position for a maximum of n bedlines, O(n). Therefore the entire algorithm runs in O(mn).
 
 Did you, at any point, exploit that our features are on single nucleotides and not larger regions?
+No, both the chromosome start and end position of the bedlines are present in the code where we compare to the query's interval. Is should be able to handle general regions.
```

Comment:
How much are you willing to bet on this? If I gave you a bed file with regions of arbitrary length and gave you a query interval of the form you have seen, would you give me all the feature regions that overlap that query?

Your code does this:

```python
if bedline.chrom_start >= int(lst[1]) and bedline.chrom_end <= int(lst[2]): 
                 print_line(bedline, args.outfile)
```

Let's say the query interval was `[5, 10)` (from five up to, but not including, 10).

Let's also say that I had features for the intervals `[1, 20)`, `[3, 6)`, `[6, 8)`, `[8, 25)`. All four intervals overlap the query, so they should all be included. But are they?

```
[1, 20) -- (bedline.chrom_start >= int(lst[1])) == (1 >= 5) == False -- not included
[3, 6)  -- (bedline.chrom_start >= int(lst[1])) == (3 >= 5) == False -- not included
[6, 8)  -- (bedline.chrom_start >= int(lst[1])) == (6 >= 5) == True
           (bedline.chrom_end <= int(lst[2])) == (8 <= 10) == True -- This one is included
[8, 25) -- (bedline.chrom_start >= int(lst[1])) == (8 >= 5) == True
           (bedline.chrom_end <= int(lst[2])) == (25 <= 10) == False -- not included
```

Your code finds 25% of the features it is supposed to here.

You don't have to fix this, but please give it a try to work out what would be necessary to find all overlapping intervals in the general case.